source/javascripts/services/app-service.ts

```diff
@@ -9,3 +9,11 @@ export const getAppSlug = (): string|null => {
 
   return null;
 };
+
+export const handleSecretAfterSave = (secret: any): void => {
```

Comment:
can we not use `any` in this case? just define a type for that in this file. Or if that's `Value` we can also port that to TS if easy enough. If that's complex, then you can just define a `type` for that which we later port to a `class`.

Reply:
Yeah, now it's definitely ugly, you are right :( I was wondering, what could be the most elegant and efficient way to create/port `Variable` to TS (the logic in `_Variable.js.erb` is pretty complex). Do you have any battle-proof suggestion?

Reply:
yeah I see that. I'm ok with creating a type from that for now, it would mean scope explosion